templates/terraform/encoders/data_catalog_tag.go.erb

```diff
@@ -0,0 +1,67 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+	// The Data Catalog Tag resource includes a map for fields where the key
+	// value is the name of the field and the value is dependent on the type
+	// of the field in the tag template. Depending on that type, a different
+	// attribute needs to be populated here, but only one of these attributes
+	// can be sent, so if an attribute is not set, we need to delete it from
+	// the object
```

Comment:
I'm not able to figure out from this why we have to read the template to see which attribute to populate rather than just doing it based off of which one was set by the user. Since our expanders check for emptiness, there shouldn't be anything in the object to delete.

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/expand_property_method.erb#L76-L82 I think if that logic is duplicated up into the "generate expanders for maps" section, that would do the trick